DaS-PC-MPChan/MainWindow.vb

```diff
@@ -365,22 +366,177 @@ Public Class MainWindow
+            blackSet.Add(n.SteamId)
+        Next
+
+        Dim candidates As New List(Of DSNode)
+        For Each t In _ircClient.ircNodes.Values
+            Dim node As DSNode = t.Item1
+            If blackSet.Contains(node.SteamId) Then Continue For
+            candidates.Add(node)
+        Next
+
+        If candidates.Count = 0 Then Return Nothing
+
+        Dim self = dsProcess.SelfNode
+
+        'These read out dsProcess memory, so don't calculate them for every node
+        Dim anorLondoInvading = self.Covenant = Covenant.DarkmoonBlade AndAlso dsProcess.HasDarkmoonRingEquiped
+        Dim forestInvading = self.Covenant = Covenant.ForestHunter AndAlso dsProcess.HasCatCovenantRingEquiped
+        Dim sorted As IOrderedEnumerable(Of DSNode) = candidates _
+            .OrderByDescending(Function(other) (other.World <> "-1--1")) _
+            .ThenByDescending(Function(other) As Boolean
+                                   'Special cross-world invasions
+                                   If anorLondoInvading Then
+                                       'TODO: Use others dark anor londo info once we have it
+                                       If other.World = AnorLondoWorld Then
+                                           Return self.canDarkmoonInvade(other)
+                                       End If
+                                   ElseIf forestInvading Then
+                                       If other.World = DarkrootGardenWorld Then
+                                           If other.HasExtendedInfo And other.Covenant = Covenant.ForestHunter Then
+                                               Return False
+                                           Else
+                                               Return self.canForestInvade(other)
+                                           End If
+                                       End If
+                                   End If
+                                   Return False
+                               End Function) _
+            .ThenByDescending(Function(other) (other.World = self.World))
+
+        Dim pvpSorting = Function(s As IOrderedEnumerable(Of DSNode))
+                             Return s.ThenByDescending(Function(other) self.canBeRedSignSummoned(other)) _
+                                     .ThenByDescending(Function(other) other.canBeRedSignSummoned(self)) _
+                                     .ThenByDescending(Function(other) other.canRedEyeInvade(self)) _
+                                     .ThenByDescending(Function(other) self.canRedEyeInvade(other)) _
+                                     .ThenByDescending(Function(other) other.canBeSummoned(self)) _
+                                     .ThenByDescending(Function(other) self.canBeSummoned(other))
+                         End Function
+
+        Dim sameMpZone = Function(other) other.MPZone = self.MPZone
+
+        If self.Covenant = Covenant.Darkwraith Then
```

Comment:
Is this functionally a "Select Case" structure or do the ElseIfs add something to it?